components/layout/fragment.rs

```diff
@@ -1275,10 +1275,18 @@ impl Fragment {
     }
 
     /// Returns true if this element can be split. This is true for text fragments, unless
-    /// `white-space: pre` is set.
+    /// `white-space: pre` or `white-space: nowrap` is set.
     pub fn can_split(&self) -> bool {
-        self.is_scanned_text_fragment() &&
-            self.style.get_inheritedtext().white_space != white_space::T::pre
+        if !self.is_scanned_text_fragment() {
+            return false;
+        }
+        match self.style.get_inheritedtext().white_space {
+            white_space::T::pre |
+            white_space::T::nowrap => false,
+            white_space::T::normal |
+            white_space::T::pre_wrap |
+            white_space::T::pre_line => true,
+        }
```

Comment:
Hmm... there are basically three useful ways to divide up the values of white-space: "are spaces preserved", "are newlines preserved", and "is wrapping allowed".  I could add helpers for all three, which might be useful for readability even if they don't have very many uses.